tempo/mlserver.py

```diff
@@ -57,6 +59,13 @@ async def _load_model(self) -> BaseModel:
 
         return model
 
+    async def _load_state(self):
+        runtime_options = self._model.runtime_options_override
+        if not runtime_options:
+            runtime_options = self._model.model_spec.runtime_options
+
+        self.state = BaseState.from_conf(runtime_options.state_options)
```

Comment:
If we were to split the options classes into multiple subclasses (e.g. `KubernetesOptions`, `DockerOptions`, `EnterpriseOptions`, etc.), would it make sense to have the `state_options` field as a member of the base abstract "`Options`" class? Or as a separate field or even a separate object?

Reply:
Not sure to be honest, I did try this just for the state class (ie a top level abs class with inheritance) but it didn't work as smoothly when parsing back from json to infer the respective class, although i think it should be doable without too much complexity